src/core/lib/security/credentials/ssl/ssl_credentials.cc

```diff
@@ -318,6 +319,7 @@ grpc_server_credentials *grpc_ssl_server_credentials_create_with_options(
   }
 
   options = *o;
+  *o = NULL;
```

Comment:
i think it's best to do this ownership transfer only once it's guaranteed to be a success.... after doing this here, we might run into an error below (`goto done`) and if we want to honor the contract with the caller that on an error, its double-pointer argument will be intact, then we will have to restore `*o = options` or something like that... which is not too clean.... so it's best to do `*o = NULL` only after we know for sure it will be a success....

again i think all this might be a style thing that might or might not be consistent with grpc code base, so maybe it's best to chat with @jboeuf before adopting :)